src/app/modules/webhook/__tests__/webhook.service.spec.ts

```diff
@@ -0,0 +1,403 @@
+    data: '{"result":"test-result"}',
+    status: 400,
+    statusText: 'failed',
+    headers: '{}',
+  },
+}
+const MOCK_WEBHOOK_DEFAULT_FORMAT_RESPONSE: Pick<
+  IWebhookResponse,
+  'response'
+> = {
+  response: {
+    data: '',
+    status: 0,
+    statusText: '',
+    headers: '',
+  },
+}
+
+describe('webhook.service', () => {
+  beforeAll(async () => await dbHandler.connect())
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  // test variables
+  let testEncryptedSubmission: IEncryptedSubmissionSchema
+  let testConfig: AxiosRequestConfig
+  let testSubmissionWebhookView: WebhookView | null
+  let testSignature: string
+
+  beforeEach(async () => {
+    jest.restoreAllMocks()
+
+    // prepare for form creation workflow
+    const MOCK_ADMIN_OBJ_ID = new ObjectID()
+    const MOCK_EPOCH = 1487076708000
+    const preloaded = await dbHandler.insertFormCollectionReqs({
+      userId: MOCK_ADMIN_OBJ_ID,
+    })
+
+    jest.spyOn(Date, 'now').mockImplementation(() => MOCK_EPOCH)
+
+    // instantiate new form and save
+    const testEncryptedForm = await FormModel.create({
+      title: 'Test Form',
+      admin: preloaded.user._id,
+      responseMode: ResponseMode.Encrypt,
+      publicKey: 'fake-public-key',
+    })
+
+    // initialise encrypted submussion
+    testEncryptedSubmission = await EncryptSubmissionModel.create({
+      form: testEncryptedForm._id,
+      authType: testEncryptedForm.authType,
+      myInfoFields: [],
+      encryptedContent: 'encrypted-content',
+      verifiedContent: 'verified-content',
+      version: 1,
+      webhookResponses: [],
+    })
+
+    // initialise webhook related variables
+    testSubmissionWebhookView = testEncryptedSubmission.getWebhookView()
+
+    testSignature = formsgSdk.webhooks.generateSignature({
+      uri: MOCK_WEBHOOK_URL,
+      submissionId: testEncryptedSubmission._id,
+      formId: testEncryptedForm._id,
+      epoch: MOCK_EPOCH,
+    })
+
+    testConfig = {
+      headers: {
+        'X-FormSG-Signature': `t=${MOCK_EPOCH},s=${testEncryptedSubmission._id},f=${testEncryptedForm._id},v1=${testSignature}`,
+      },
+      maxRedirects: 0,
+    }
+  })
+
+  describe('saveWebhookRecord', () => {
+    it('should return transform mongo error if database update for webhook fails', async () => {
+      // Arrange
+      const mockWebhookResponse = {
+        ...MOCK_WEBHOOK_SUCCESS_RESPONSE,
+        signature: testSignature,
+        webhookUrl: MOCK_WEBHOOK_URL,
+      } as IWebhookResponse
+
+      const mockDBError = new Error(DEFAULT_ERROR_MSG)
+
+      jest
+        .spyOn(EncryptSubmissionModel, 'addWebhookResponse')
+        .mockRejectedValueOnce(mockDBError)
+
+      // Act
+      const actual = await saveWebhookRecord(
+        testEncryptedSubmission._id,
+        mockWebhookResponse,
+      )
+
+      // Assert
+      const expectedError = transformMongoError(mockDBError)
+
```

Comment:
should expect `addWebhookResponse` to be called with the correct arguments. this is the purpose of swapping it out for a spy function

Reply:
this applies to all the tests